src/repositories/users.py

```diff
@@ -1,24 +1,33 @@
 # ------- REPOSITORY FILE -------
-from ..models import UserBase, UserCreate
+from ..models import UserCreate, UserResponse, UserWithRoleResponse
 
 
 class UsersRepository:
     def __init__(self, db_session):
         self.db_session = db_session
 
-    async def get_user_by_email(self, email: str) -> UserBase:
+    async def get_user_by_email(self, email: str) -> UserResponse:
         async with self.db_session.transaction():
             query = "SELECT * FROM d_roles.users WHERE email = :email"
             return await self.db_session.fetch_one(query, {"email": email})
 
-    async def add_user(self, user: UserCreate) -> int:
+    async def add_user(self, user: UserCreate) -> UserResponse:
         async with self.db_session.transaction():
-            query = "INSERT INTO d_roles.users (email, is_email_confirmed) VALUES (:email, :is_email_confirmed) RETURNING id"
+            query = "INSERT INTO d_roles.users (email, is_email_confirmed) VALUES (:email, :is_email_confirmed) RETURNING *"
             values = {"email": user.email, "is_email_confirmed": False}
-            new_user_id = await self.db_session.execute(query, values)
-            return new_user_id
+            return await self.db_session.fetch_one(query, values)
 
-    async def get_user_by_id(self, user_id: int) -> UserBase:
+    async def get_user_by_id(self, user_id: int) -> UserResponse:
         async with self.db_session.transaction():
             query = "SELECT * FROM d_roles.users WHERE id = :id"
             return await self.db_session.fetch_one(query, {"id": user_id})
+
+    async def get_users_by_team_id(self, team_id: int) -> list[UserWithRoleResponse]:
```

Comment:
```suggestion
    async def get_users_by_team_id(self, team_id: int) -> list[UserWithRoleResponse] | None:
```
In case there are no users in the team yet?

Reply:
Then it should be an empty list

Reply:
I am not a fan of type mutation. Either return an empty array or a 404 not found this ensure `object.length` or `len(response)` wont raise an exception which is a pain for the dev who will use it